bugsnag-plugin-android-ndk/src/test/cpp/migrations/EventMigrationV6Tests.cpp

```diff
@@ -0,0 +1,150 @@
+  event->app.version_code = 57;
+
+  // breadcrumbs
+  auto max = 50;
+  event->crumb_first_index = 2; // test the circular buffer logic
+  char name[30];
+  for (int i = event->crumb_first_index; i < max; i++) {
+    sprintf(name, "mission %d", i - event->crumb_first_index);
+    insert_crumb(event->breadcrumbs, i, name, BSG_CRUMB_STATE, 1638992630014,
+                 "Now we know what they mean by 'advanced' tactical training.");
+  }
+  for (int i = 0; i < event->crumb_first_index; i++) {
+    sprintf(name, "mission %d", (max - event->crumb_first_index) + i);
+    insert_crumb(event->breadcrumbs, i, name, BSG_CRUMB_STATE, 1638992630014,
+                 "Now we know what they mean by 'advanced' tactical training.");
+  }
+  event->crumb_count = max;
+
+  // device
+  event->device.cpu_abi_count = 1;
+  strcpy(event->device.cpu_abi[0].value, "mipsx");
+  strcpy(event->device.id, "ffffa");
+  event->device.jailbroken = true;
+  strcpy(event->device.locale, "en_AU#Melbun");
+  strcpy(event->device.manufacturer, "HI-TEC™");
+  strcpy(event->device.model, "🍨");
+  strcpy(event->device.orientation, "sideup");
+  strcpy(event->device.os_name, "BOX BOX");
+  strcpy(event->device.os_version, "98.7");
+  { // -- runtime versions
+    strcpy(event->device.os_build, "beta1-2");
+    event->device.api_level = 32;
+  }
+  event->device.time = 1638992630;
+  event->device.total_memory = 3278623;
+
+  // exceptions
+  strcpy(event->error.errorClass, "SIGBUS");
+  strcpy(event->error.errorMessage, "POSIX is serious about oncoming traffic");
+  strcpy(event->error.type, "C");
+  event->error.frame_count = 2;
+  event->error.stacktrace[0].frame_address = 454379;
+  event->error.stacktrace[0].load_address = 2367523;
+  event->error.stacktrace[0].symbol_address = 776;
+  strcpy(event->error.stacktrace[0].method, "makinBacon");
+  strcpy(event->error.stacktrace[0].filename, "lib64/libfoo.so");
+  event->error.stacktrace[1].frame_address = 342334; // will become method hex
+
+  // metadata
+  strcpy(event->app.active_screen, "Menu");
+  bugsnag_event_add_metadata_bool(event, "metrics", "experimentX", false);
+  bugsnag_event_add_metadata_string(event, "metrics", "subject", "percy");
+  bugsnag_event_add_metadata_string(event, "app", "weather", "rain");
+  bugsnag_event_add_metadata_double(event, "metrics", "counter", 47.5);
+
+  // session info
+  event->handled_events = 5;
+  event->unhandled_events = 2;
+  strcpy(event->session_id, "aaaaaaaaaaaaaaaa");
+  strcpy(event->session_start, "2031-07-09T11:08:21+00:00");
+
+  // user
+  strcpy(event->user.email, "fenton@io.example.com");
+  strcpy(event->user.name, "Fenton");
+  strcpy(event->user.id, "fex01");
+
+  return event;
+}
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+JNIEXPORT jstring JNICALL
+Java_com_bugsnag_android_ndk_migrations_EventMigrationV6Tests_migratePayloadInfo(
+    JNIEnv *env, jobject _this, jstring temp_file) {
+  const char *path = (*env).GetStringUTFChars(temp_file, nullptr);
```

Comment:
I'm guessing there's no corresponding `ReleaseStringUTFChars` because this is a test?

Reply:
pretty much - its short lived either way :)